service/src/test/java/org/cbioportal/service/impl/UuidDataAccessTokenServiceImplTest.java

```diff
@@ -0,0 +1,182 @@
+
+@TestPropertySource(
+    properties = { "jwt.secret_key = +NbopXzb/AIQNrVEGzxzP5CF42e5drvrXTQot3gfW/s=",
+                    "dat.max_number_per_user = 1",
+                    "dat.revoke_other_tokens = true",
+                    "dat.ttl_seconds = 2",
+                    "jwt.issuer = cbioportal.org"
+    },
+    inheritLocations = false
+)
+@ContextConfiguration(classes=UuidDataAccessTokenServiceImplTestConfiguration.class)
+@RunWith(SpringRunner.class)
+public class UuidDataAccessTokenServiceImplTest {
+
+    @Autowired
+    private UuidDataAccessTokenServiceImplTestConfiguration uuidDataAccessTokenServiceImplTestConfiguration;
+
+    @Autowired
+    private DataAccessTokenRepository dataAccessTokenRepository;
+
+    @Autowired
+    @Qualifier("uuidDataAccessTokenServiceImpl")
+    private UuidDataAccessTokenServiceImpl uuidDataAccessTokenServiceImpl;
+
+    @Value("${dat.ttl_seconds}") 
+    private int datTtlSeconds;
+    
+    /* Test for checking exception thrown when token limit reached
+     */
+    @Test(expected = MaxNumberTokensExceededException.class)
+    public void testNonAutoExpireCreateTokenWhenLimitReached() {
+        DataAccessToken newDataAccessToken = uuidDataAccessTokenServiceImpl.createDataAccessToken(UuidDataAccessTokenServiceImplTestConfiguration.MOCK_USERNAME_WITH_ONE_TOKEN, false);
+    }
+
+    /* Test for creating a token when autoexpire is on
+     * tests that new token is created/MaxNumberTokensExcceededException is not thrown
+     * deletedToken should be the oldest token
+     */
+    @Test
+    public void testAutoExpireCreateTokenWhenLimitReached() {
+        // Testing for service with limit 1 token
+        uuidDataAccessTokenServiceImplTestConfiguration.resetAddedDataAccessToken();
+        uuidDataAccessTokenServiceImplTestConfiguration.resetDeletedDataAccessToken();
+        DataAccessToken newDataAccessToken = uuidDataAccessTokenServiceImpl.createDataAccessToken(UuidDataAccessTokenServiceImplTestConfiguration.MOCK_USERNAME_WITH_ONE_TOKEN, true);
+        Date expectedExpirationDate = getExpectedExpirationDate();
+        String deletedDataAccessToken = uuidDataAccessTokenServiceImplTestConfiguration.getDeletedDataAccessToken();
+        DataAccessToken createdDataAccessToken = uuidDataAccessTokenServiceImplTestConfiguration.getAddedDataAccessToken();
+        if (createdDataAccessTokenWithWrongInformation(createdDataAccessToken, UuidDataAccessTokenServiceImplTestConfiguration.MOCK_USERNAME_WITH_ONE_TOKEN, expectedExpirationDate)) {
+            Assert.fail("Created token (Username: " + createdDataAccessToken.getUsername() + ", Expiration: " + createdDataAccessToken.getExpiration() + ") differs from expected (Username: " + UuidDataAccessTokenServiceImplTestConfiguration.MOCK_USERNAME_WITH_ONE_TOKEN + ", Expiration: " + expectedExpirationDate.toString() + ")");
+        }
+        if (deletedDataAccessToken != uuidDataAccessTokenServiceImplTestConfiguration.OLDEST_TOKEN_UUID) {
```

Comment:
here we make sure the oldest token is the one which expires first. Technically, we do not expire the oldest token (the one with the earliest creation date) ... we expire the one soonest to become expired (the earliest expiration date). But maybe this distinction is not important.